src/Ticket.php

```diff
@@ -4379,7 +4379,12 @@ public function showForm($ID, array $options = [])
                         break;
                 }
                 // Check category / entity validity
-                if (!in_array($cat->fields['entities_id'], getSonsOf('glpi_entities', $options['entities_id']))) {
+                if (
+                    !in_array($cat->fields['entities_id'], array_merge(
+                        getSonsOf('glpi_entities', $options['entities_id']),
```

Comment:
I am not sure to understand this `getSonsOf('glpi_entities', $options['entities_id'])`.

It mean that we would agree to create a ticket in the root entity with a category that is present only in a child entity ?

Reply:
Yes, you're right. In the latest commit, I adapted your previous suggestion to fix the original issue #17811

Since we don’t have E2E tests on BF, I’m not sure how to implement an automated test for this. (It works in manual testing, though.)

Reply:
Or move the entity check code into a method and test the method directly, which might be simpler to understand.